test_suite/stdlib.jsonnet

```diff
@@ -561,6 +561,19 @@ local some_json = {
   '"': null,
 };
 
+local bare_yaml_test = {
+  abc: true,
```

Comment:
Many more tests are needed. Starting with all possible capitalization of all keywords, stuff that isn't even a valid value and breaks the syntax (e.g. `-`), various floating point formats, hex numbers, 
datetimes, whitespace, empty strings etc.

Reply:
I went through the core types of the yaml spec (https://yaml.org/type/) and made sure that everything is quoted, taking extra care to error on the side of unnecessary quotation rather than the opposite.  To help make sure nothing slips by, I have added a large number of test cases to demonstrate this.

All of the types you listed here are covered, along with several others documented in the spec.